client/python/examples/general.py

```diff
@@ -0,0 +1,173 @@
+"""
+A more fledged out version of `simple.py` where we create a queue only
+if it doesn't exist, and then create a jobset and job, and wait until
+the job succeeds or fails.
+"""
+
+import os
+import time
+import uuid
+
+import grpc
+from armada_client.client import ArmadaClient
+from armada_client.k8s.io.api.core.v1 import generated_pb2 as core_v1
+from armada_client.k8s.io.apimachinery.pkg.api.resource import (
+    generated_pb2 as api_resource,
+)
+from armada_client.typings import EventType
+
+
+def create_dummy_job(client: ArmadaClient):
+    """
+    Create a dummy job with a single container.
+    """
+
+    # For infomation on where this comes from,
+    # see https://github.com/kubernetes/api/blob/master/core/v1/generated.proto
+    pod = core_v1.PodSpec(
+        containers=[
+            core_v1.Container(
+                name="container1",
+                image="index.docker.io/library/ubuntu:latest",
+                args=["sleep", "10s"],
+                securityContext=core_v1.SecurityContext(runAsUser=1000),
+                resources=core_v1.ResourceRequirements(
+                    requests={
+                        "cpu": api_resource.Quantity(string="120m"),
+                        "memory": api_resource.Quantity(string="510Mi"),
+                    },
+                    limits={
+                        "cpu": api_resource.Quantity(string="120m"),
+                        "memory": api_resource.Quantity(string="510Mi"),
+                    },
+                ),
+            )
+        ],
+    )
+
+    return [client.create_job_request_item(priority=1, pod_spec=pod)]
+
+
+def wait_for_job_event(client, event_stream, job_id: str, event_state: EventType):
+    """
+    Wait for a job event to occur.
+
+    Will automatically return if the job fails or the event stream is closed.
+    """
+
+    # Contains all the possible message types
+    for event in event_stream:
+
+        event = client.unmarshal_event_response(event)
+
+        if event.message.job_id == job_id:
+            if event.type == EventType.failed:
```

Comment:
There are a few other events you are missing.  (Cancelled, DuplicatedFound). You can look at the events code in go to see what we consider terminal.